third_party/move/tools/move-mutator/src/compiler.rs

```diff
@@ -306,6 +310,51 @@ pub fn verify_mutant(
     Ok(())
 }
 
+/// Rewrite the manifest file to use absolute paths.
+///
+/// # Arguments
+///
+/// * `root` - the path to the package root.
+/// * `tempdir` - the path to the temporary directory.
+///
+/// # Errors
+///
+/// * If any error occurs during the rewrite, the appropriate error is returned using anyhow.
+///
+/// # Returns
+///
+/// * `Result<(), anyhow::Error>` - Ok if the rewrite is successful, or an error if any error occurs.
+fn rewrite_manifest_for_mutant(root: &Path, tempdir: &Path) -> Result<(), anyhow::Error> {
+    let mut manifest_string = fs::read_to_string(root.join(SourcePackageLayout::Manifest.path()))?;
+    let manifest = manifest_parser::parse_move_manifest_string(manifest_string.clone())?;
+    let manifest = manifest_parser::parse_source_manifest(manifest)?;
+    let curdir = std::env::current_dir()?;
+
+    // We need to switch to package dir as paths in manifest are relative to package dir.
+    std::env::set_current_dir(root)?;
+
+    manifest
+        .dependencies
+        .values()
+        .into_iter()
+        .chain(manifest.dev_dependencies.values())
+        .for_each(|dep| {
+            let dep_canon = dep.local.canonicalize();
+            if let Ok(dep_canon) = dep_canon {
+                manifest_string = manifest_string
+                    .replace(dep.local.to_str().unwrap(), dep_canon.to_str().unwrap());
+            }
+        });
+
+    // Switch back to the original dir.
+    std::env::set_current_dir(curdir)?;
+    fs::write(
+        tempdir.join(SourcePackageLayout::Manifest.path()),
+        manifest_string,
+    )?;
```

Comment:
A simple swap, since setting the curr dir back is like a cleanup part at the end
```suggestion

    fs::write(
        tempdir.join(SourcePackageLayout::Manifest.path()),
        manifest_string,
    )?;
    
    // Switch back to the original dir.
    std::env::set_current_dir(curdir)?;
```

Reply:
Yes, however I thought about it differently.

As `tempdir` is always absolute path, so the current directory has no meaning for it. But, if something goes wrong writing this particular mutant and we'd like to try with the next one, we'll be not in the original directory.

That's why I got back to the original dir ASAP, to have no operation that could fail between changing dir and going back.

Maybe I'll add a comment to make it clear in the code?

Reply:
I don't think it's not need - it just how I load function in my mind 😄